app-services/secrets/README.md

```diff
@@ -1,36 +1,116 @@
+       
+   ```
+
+3. The default configuration is for running natively and we use environment overrides in the compose files for use in docker.
+
+4. Add the secrets example application service to your EdgeX docker compose file. This includes the environment overrides for the SecretStore and SecretStoreExclusive configurations:
+
+   ```yml
+     app-service-secrets-example:
+       image: edgexfoundry/docker-secrets-example:dev
+       container_name: app-service-secrets-example
+       hostname: app-service-secrets-example
+       ports:
+         - "48095:48095"
+       networks:
+         - edgex-network
+       environment:
+         <<: *common-variables
+         SecretStore_Protocol: https
+         SecretStore_TokenFile: /tmp/edgex/secrets/edgex-appservice/secrets-token.json
+         SecretStoreExclusive_Host: edgex-vault
+         SecretStoreExclusive_Protocol: https
+         SecretStoreExclusive_RootCaCertPath: /tmp/edgex/secrets/ca/ca.pem
+         SecretStoreExclusive_ServerName: edgex-vault
+         SecretStoreExclusive_TokenFile: /tmp/edgex/secrets/secrets-example/secrets-token.json
+       volumes:
+         - /tmp/edgex/secrets/ca:/tmp/edgex/secrets/ca:ro,z
+         - /tmp/edgex/secrets/edgex-appservice:/tmp/edgex/secrets/edgex-appservice:ro,z 
+         - /tmp/edgex/secrets/secrets-example:/tmp/edgex/secrets/secrets-example:ro,z 
+       depends_on:
+         - vault-worker
+   ```
+
+   4. Run EdgeX security services in docker:
 
-Application Services can retrieve their secrets from the underlying secret store using the GetSecrets() API in the SDK. 
+      The secrets example in docker is dependent on the EdgeX security services, at a minimum. 
 
-`.GetSecrets(path string, keys ...string)` is used to retrieve secrets from the secret store. `path` describes the type or location of the secrets to retrieve. If specified it is appended to the base path from the secret store configuration. `keys` specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the specified path will be returned.
+      1. Run the secret store services in EdgeX using:
 
-#### Secure mode
+         `docker-compose -f <docker-compose file name> up -d vault-worker`
 
-If in secure mode, the secrets are stored and retrieved from Vault based on the SecretStore configuration values.
+         Verify that the following services are running:
 
-Setup the application config to use  Vault's secrets engine:
+         ```
+         edgex-core-consul
+         edgex-files      
+         edgex-secrets-setup
+         edgex-vault        
+         edgex-vault-worker
+         ```
+
+      2. Wait for about a minute to ensure tokens are created for the services that were specified in the docker compose section of vault-worker service for `ADD_SECRETSTORE_TOKENS` environment variable. Verify that the tokens for `edgex-appservice` and `secrets-example` exist under `/tmp/edgex/secrets`.
```

Comment:
remove `edgex-appservice and` or change `edgex-appservice` to `edgex-application-service`